algebra/_common/kkt.c

```diff
@@ -201,44 +335,37 @@ csc* form_KKT(c_float*    P_x,
 
 #if EMBEDDED != 1
 
-void update_KKT_P(csc         *KKT,
-                  c_float*    P_x,
-                  c_int*      P_p,
-                  c_int       n,
-                  c_int  *PtoKKT,
-                  c_float param1,
-                  c_int  *Pdiag_idx,
-                  c_int   Pdiag_n) {
+void update_KKT_P(csc*        KKT,
+                  csc*        P,
+                  c_int*      PtoKKT,
+                  c_float     param1,
+                  c_int       format) {
 
-  c_int i, j; // Iterations
+  c_int j; // Iterations
 
   // Update elements of KKT using P
-  for (i = 0; i < P_p[n]; i++) {
-    KKT->x[PtoKKT[i]] = P_x[i];
+  for (j = 0; j < P->p[P->n]; j++) {
```

Comment:
You may want to define `c_int nnzP = P->p[P->n];` and use `nnzP` in the loop condition instead.